gateway/integration_tests/src/main.rs

```diff
@@ -199,284 +199,289 @@ async fn mnemonic_upgrade_test(
-        assert!(
-            !mnemonic_response
-                .legacy_federations
-                .contains(&federation_id)
-        );
-        assert_eq!(mnemonic_response.legacy_federations.len(), 0);
+            let data_dir: PathBuf = env::var(FM_DATA_DIR_ENV)
+                .expect("Data dir is not set")
+                .parse()
+                .expect("Could not parse data dir");
+            let gw_fed_db = data_dir.join("lnd").join(format!("{federation_id}.db"));
+            remove_dir_all(gw_fed_db)?;
 
-        info!("Verified deleting database will migrate the federation to use mnemonic");
+            gw_lnd.connect_fed(fed).await?;
 
-        info!("Successfully completed mnemonic upgrade test");
+            // Verify that the re-connected federation is not a legacy federation
+            let mnemonic_response = gw_lnd.get_mnemonic().await?;
+            assert!(
+                !mnemonic_response
+                    .legacy_federations
+                    .contains(&federation_id)
+            );
+            assert_eq!(mnemonic_response.legacy_federations.len(), 0);
 
-        Ok(())
-    })
-    .await
+            info!("Verified deleting database will migrate the federation to use mnemonic");
+
+            info!("Successfully completed mnemonic upgrade test");
+
+            Ok(())
+        })
+        .await
 }
 
 /// Test that sets and verifies configurations within the gateway
 #[allow(clippy::too_many_lines)]
 async fn config_test(gw_type: LightningNodeType) -> anyhow::Result<()> {
-    Box::pin(devimint::run_devfed_test(
-        |dev_fed, process_mgr| async move {
-            let gatewayd_version = util::Gatewayd::version_or_default().await;
-            if gatewayd_version < *VERSION_0_5_0_ALPHA && gw_type == LightningNodeType::Ldk {
-                return Ok(());
-            }
-
-            let gw = match gw_type {
-                LightningNodeType::Lnd => dev_fed.gw_lnd_registered().await?,
-                LightningNodeType::Ldk => dev_fed.gw_ldk_connected().await?,
-            };
-
-            // Try to connect to already connected federation
-            let invite_code = dev_fed.fed().await?.invite_code()?;
-            let output = cmd!(gw, "connect-fed", invite_code.clone())
-                .out_json()
-                .await;
-            assert!(
-                output.is_err(),
-                "Connecting to the same federation succeeded"
-            );
-            info!("Verified that gateway couldn't connect to already connected federation");
-
-            let gatewayd_version = util::Gatewayd::version_or_default().await;
-
-            // Change the routing fees for a specific federation
-            let fed_id = dev_fed.fed().await?.calculate_federation_id();
-            gw.set_federation_routing_fee(fed_id.clone(), 20, 20000)
-                .await?;
+    Box::pin(
+        devimint::run_devfed_test()
+            .num_feds(2)
```

Comment:
All of this so this one test can allocate 2 Fedimints. :shrug: 

But I am planning to make other pieces of devimint configurable too, so we can stop wasting resources spawning everything all the time.

Reply:
Haha sorry, I do think this is important to test though :) I think the improvement, also will be cool not to spawn everything each test

Reply:
Perhaps some tests could even skip channel openings, which has traditionally been a flaky part of devimint